include/ck/tensor_operation/gpu/device/impl/device_gemm_xdl_cshuffle_v3r1.hpp

```diff
@@ -247,19 +248,18 @@ struct DeviceGemm_Xdl_CShuffleV3R1 : public DeviceGemmV2R1<ALayout,
 
             auto reduce = DeviceReduceInstance{};
 
-            auto argument_ptr =
-                reduce.MakeArgumentPointer(in_lengths,
-                                           in_strides,
-                                           DsLengths,
-                                           DsStrides,
-                                           out_lengths,
-                                           out_strides,
-                                           reduce_dims,
-                                           arg.IsReduceAdd() ? arg.p_workspace_ : arg.p_c_grid,
-                                           arg.p_ds,
-                                           arg.p_c_grid,
-                                           PassThrough{},
-                                           OutElementwiseOperation{});
+            auto argument_ptr = reduce.MakeArgumentPointer(in_lengths,
+                                                           in_strides,
+                                                           DsLengths,
+                                                           DsStrides,
+                                                           out_lengths,
+                                                           out_strides,
+                                                           reduce_dims,
+                                                           arg.p_workspace_,
+                                                           arg.p_ds,
```

Comment:
`arg.IsReduceAdd() ? arg.p_workspace_ : arg.p_c_grid` is gone?

Reply:
multi-D and reduce both  use workspace now